src/controller.coffee

```diff
@@ -184,6 +186,15 @@ class Controller extends EventEnabled
   processResponse: (response) ->
     @processors.reduce ((data, fn) -> fn data), response
 
+  ###*
+   * Transform the params by passing it through a set of data  paramsProcessors,
+   * if any.
+   *
+   * @param  {Object} The params passed to the query.
+  ###
+  prePreprocessParams: (params) ->
+    @paramsPreprocessors.reduce ((data, fn) -> fn data), params
+
```

Comment:
You have a refactoring opportunity here. This is essentially the same as `processResponse`. You could rename `processResponse(response)` to `transformObject(obj, transformations)` and provide `transformations` array as a parameter.

You would use it as:

```coffee
@transformObject(response, @processors)
@transformObject(@params, @paramsPreprocessors)
```

Reply:
nice catch, done